docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+The purpose of the IBC connection handshake protocol, at the highest level of abstraction, is to enable two chains to establish a connection with each other [[ICS 003](#references)].
+Throughout the rest of this document, we will often use the abbreviation "CH" to stand for "connection handshake" problem (or protocol, depending on the context).
+
+#### Sequential problem statement
+
+Instead of _chains_, at this abstract level of specification we can reason in terms of _processes_.
+There are two processes in our problem statement, which we can label explicitly as an _initiator_ (or source) process and a _responder_ (or destination) process.
+
+
+###### Problem statement and guarantees
+A connection handshake allows, briefly, that two processes agree on a consistent view of their state, and each process allocates a new connection associated with this state.
+We identify the following requirements that the connection handshake should guarantee:
+
+- **[CH/Seq/1/Agreement]** If the initiator finishes the handshake, establishing a new connection `ci` with the responder, and the responder finishes the handshake obtaining the connection `cr`, then `ci == cr`.
+In other words, the two ends of the connection (one at the initiator and the other at the responder) are consistent.
+
+
+- **[CH/Seq/2/Termination]** Eventually, the two processes should both finish the handshake, establishing a connection.
+
+Requirement **[CH/Seq/1/Agreement]** represents the basic safety guarantee, while **[CH/Seq/2/Termination]** is the basic liveness guarantee we would like to have.
+
+The diagram below sketches the high-level problem statement.
+
+```
+                        IBC Connection
+     +-----------+    Handshake Protocol    +-----------+
+     |           |<-------- step 1 -------->|           |
+     | Initiator |    (execute handshake)   | Responder |
+     |           |                          |           |
+     |          _| (connection established) |_          |
+     |         |c|<------------------------>|c|         |
+     |          ▔|                          |▔          |
+     +-----------+                          +-----------+
+```
+
+
+## L2. Protocol specification / protocol view
+
+In this section we attempt to make the problem specification more concrete.
+We define what the processes are, their fault-model, as well as communication and synchrony assumptions.
+
+### 2.1 System model specification
+
+#### System model
+
+A **chain** is a state machine.
+Each chain may be replicated, i.e., consisting of a replicated state machine (RSM), or it may be a standalone process.
+In practice, there are certain requirements which each chain must satisfy (e.g., see [[ICS 024](#references)]).
+At this level of specification, however, we can assume each chain is a single **process**.
+
+The connection handshake protocol is asymmetric, meaning that the two chains that interact play different roles within the protocol.
```

Comment:
We should be careful here as every module (on corresponding chain) can act as both initiator and responder. Although protocol is asymmetric we cannot necessarily define a priori who is playing what role. In addition, there is potentially a problematic case in which both modules at the same time trigger connection handshake (act as initiator) and we want to formalise this case and talk about guarantees we provide in that case.

Reply:
Indeed, there are some tricky cases to consider there. I tried to rule out some of these corner-cases with the property **[CH/Proto/1/Uniqueness]**. As long as connections have different (unique) identifiers, they can proceed concurrently and should not matter who was initiator or responder.